pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSetMetaData.java

```diff
@@ -206,10 +212,14 @@ private void fetchFieldMetaData() throws SQLException {
     }
 
     // see if cached
-    String idFields = _cache.getIdFields(fields);
-    if (_cache.isCached(idFields)) {
+    final String idFields = createCacheKey(fields);
```

Comment:
Frankly speaking, I would like to have this refactored:  can you please switch to `(tableOid, position) -> Field` kind of cache?

In other words, we'd better cache the same field only once, not multiple times for different variations of result rows